src/go/rpk/pkg/cli/cluster/health.go

```diff
@@ -72,13 +72,25 @@ following conditions are met:
 func printHealthOverview(hov *adminapi.ClusterHealthOverview) {
 	types.Sort(hov)
 	out.Section("CLUSTER HEALTH OVERVIEW")
-	overviewFormat := `Healthy:                     %v
-Unhealthy reasons:           %v
-Controller ID:               %v
-All nodes:                   %v
-Nodes down:                  %v
-Leaderless partitions:       %v
-Under-replicated partitions: %v
-`
-	fmt.Printf(overviewFormat, hov.IsHealthy, hov.UnhealthyReasons, hov.ControllerID, hov.AllNodes, hov.NodesDown, hov.LeaderlessPartitions, hov.UnderReplicatedPartitions)
+
+	// leaderless partitions and under-replicated counts are available starting
+	// v23.3.
+	lp := "Leaderless partitions:"
+	urp := "Under-replicated partitions:"
+	if hov.LeaderlessCount != nil {
+		lp = fmt.Sprintf("Leaderless partitions (%v):", *hov.LeaderlessCount)
```

Comment:
Putting the count in parentheses before the colon is a bit confusing, but I don't have a great alternative. @travisdowns wdyt / is this fine?

Reply:
fwiw, I took the idea from here: https://github.com/redpanda-data/redpanda/issues/11378#issuecomment-1688787212

Reply:
I mean I suggested this approach so I guess I have to say it's fine, but I would lean on you guys to know the best DX. I think a separate line for the count is fine too.

One other question is if we should give an explicit indicator that the list is truncated (when it is). Just to avoid the future question of "the count says 200 but the list only has 128 entries in it".

Reply:
@travisdowns added a new commit https://github.com/redpanda-data/redpanda/pull/13553/commits/d02d6b2e47113909c8f6fab01b28c9f93f44d733 which adds a "...truncated" message at the end of the list if the count > len(list). :+1: